pkg/ipcache/ipcache.go

```diff
@@ -580,6 +580,23 @@ func (ipc *IPCache) LookupByIdentity(id identity.NumericIdentity) (ips []string)
 	return ips
 }
 
+// LookupByHostRLocked returns the list of IPs returns the set of IPs
+// (endpoint or CIDR prefix) that have hostIP associated as the host of the
+// entry. Requires the caller to hold the RLock.
+func (ipc *IPCache) LookupByHostRLocked(hostIP net.IP) (cidrs []net.IPNet) {
```

Comment:
We can cut the lookup by half if we support sending a v4 and a v6 hostIP addresses as an argument of this function.